File: src/main/java/com/klu/service/StudentService.java
package com.klu.service;

import com.klu.model.Student;
import java.util.List;

public interface StudentService {
    Student    addStudent(Student student);
    List<Student> getAllStudents();
    Student    updateStudent(Long id, Student student);
    void       deleteStudent(Long id);
}

File: src/components/StudentService.js
import axios from 'axios';


const API_BASE = 'http://localhost:8080/students';


export const getAllStudents = () => axios.get(API_BASE);

export const addStudent = (student) => axios.post(API_BASE, student);


export const updateStudent = (id, student) => axios.put(`${API_BASE}/${id}`, student);


export const deleteStudent = (id) => axios.delete(`${API_BASE}/${id}`);
